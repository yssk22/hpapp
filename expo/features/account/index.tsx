import ListItemClearCache from './internal/ListItemClearCache';

export { ListItemClearCache };
