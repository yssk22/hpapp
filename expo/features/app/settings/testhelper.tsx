import { act, render } from '@testing-library/react-native';

import { SettingsProvider, SettingsProviderProps } from './';
import SettingsList from './internals/SettingsList';

/**
 * Clear all settings
 */
export async function clearSettings() {
  await Promise.all(
    SettingsList.map((settings) => {
      settings.clear();
    })
  );
}

export async function renderSettingsProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  await clearSettings();
  const content = render(<SettingsProvider {...props}>{children}</SettingsProvider>);
  await act(async () => {});
  return content;
}
