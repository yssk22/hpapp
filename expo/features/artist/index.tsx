import ArtistMemberFollowIcon from './internals/ArtistMemberFollowIcon';
import ArtistMemberIcon, { ArtistMemberIconSize } from './internals/ArtistMemberIcon';

export { ArtistMemberFollowIcon, ArtistMemberIcon, ArtistMemberIconSize };
