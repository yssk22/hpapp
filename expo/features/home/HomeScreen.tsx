import { defineScreen } from '@hpapp/features/common/stack';

import HomeTab from './internals/HomeTab';

export default defineScreen('/', function HomeScreen() {
  return <HomeTab />;
});
