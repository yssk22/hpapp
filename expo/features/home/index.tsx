import { HomeTabSection } from './internals/types';

export { HomeTabSection };
