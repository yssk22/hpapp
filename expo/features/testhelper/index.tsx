import {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered
} from './internals/jest';
import { ScreenDecorator } from './internals/storybook';

export {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered,
  ScreenDecorator
};
