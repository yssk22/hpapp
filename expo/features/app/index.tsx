import AppRelayProvider from './internals/AppRelayProvider';
import { default as Root, AppRootProps as RootProps } from './internals/AppRoot';
import AppUpdateBanner from './internals/AppUpdateBanner';
import { ThemeProvider, ThemeProviderProps } from './theme';

export type AppRootProps = RootProps & ThemeProviderProps;

export function AppRoot({ screens, ...props }: AppRootProps) {
  return (
    <ThemeProvider {...props}>
      <AppRelayProvider>
        <>
          <AppUpdateBanner />
          <Root screens={screens} />
        </>
      </AppRelayProvider>
    </ThemeProvider>
  );
}
