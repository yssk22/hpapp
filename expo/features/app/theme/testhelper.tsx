import { SettingsProviderProps } from '@hpapp/features/app/settings';
import { renderSettingsProvider } from '@hpapp/features/app/settings/testhelper';

import { ThemeProvider } from './';

export async function renderThemeProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  return renderSettingsProvider(<ThemeProvider>{children}</ThemeProvider>, props);
}
