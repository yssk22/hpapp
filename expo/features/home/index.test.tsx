import { renderUserScreen } from '@hpapp/features/testhelper';
import { sleep } from '@hpapp/foundation/globals';
import { act, waitFor } from '@testing-library/react-native';

import HomeScreen from './HomeScreen';

describe('home', () => {
  test('HomeScreen', async () => {
    const content = await renderUserScreen(HomeScreen);
    await waitFor(async () => content.queryByTestId('HomeTabFeedSectionList'));
    await act(async () => {
      await sleep(500); // Wait for the tabs to be rendered
    });
    expect(content).toMatchSnapshot();
  });
});
