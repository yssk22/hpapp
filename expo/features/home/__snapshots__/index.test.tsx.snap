// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home HomeScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <RNSScreenNavigationContainer
                style={
                  {
                    "flex": 1,
                    "overflow": "hidden",
                  }
                }
              >
                <RNSScreen
                  activityState={2}
                  collapsable={false}
                  gestureResponseDistance={
                    {
                      "bottom": -1,
                      "end": -1,
                      "start": -1,
                      "top": -1,
                    }
                  }
                  onGestureCancel={[Function]}
                  sheetAllowedDetents="large"
                  sheetCornerRadius={-1}
                  sheetExpandsWhenScrolledToEdge={true}
                  sheetGrabberVisible={false}
                  sheetLargestUndimmedDetent="all"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "zIndex": 0,
                    }
                  }
                >
                  <View
                    accessibilityElementsHidden={false}
                    importantForAccessibility="auto"
                    style={
                      [
                        {
                          "backgroundColor": "white",
                          "flex": 1,
                        },
                        [
                          {
                            "flex": 1,
                            "flexDirection": "column-reverse",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "black",
                              },
                              {
                                "fontSize": 14,
                              },
                              undefined,
                            ]
                          }
                          testID="HomeTabFeedSectionList"
                        >
                          SectionList
                        </Text>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": 0,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            {
                              "backgroundColor": "#0075c2",
                              "borderBottomColor": "white",
                              "flex": 1,
                              "shadowColor": "white",
                              "shadowOffset": {
                                "height": 0.5,
                                "width": 0,
                              },
                              "shadowOpacity": 0.85,
                              "shadowRadius": 0,
                            }
                          }
                        />
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          {
                            "height": 44,
                            "maxHeight": undefined,
                            "minHeight": undefined,
                            "opacity": undefined,
                            "transform": undefined,
                          }
                        }
                      >
                        <View
                          pointerEvents="none"
                          style={
                            {
                              "height": 0,
                            }
                          }
                        />
                        <View
                          pointerEvents="box-none"
                          style={
                            {
                              "alignItems": "stretch",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            pointerEvents="box-none"
                            style={
                              {
                                "alignItems": "flex-start",
                                "flexBasis": 0,
                                "flexGrow": 1,
                                "justifyContent": "center",
                                "marginStart": 0,
                              }
                            }
                          />
                          <View
                            collapsable={false}
                            pointerEvents="box-none"
                            style={
                              {
                                "justifyContent": "center",
                                "marginHorizontal": 16,
                                "maxWidth": 718,
                              }
                            }
                          >
                            <Text
                              accessibilityRole="header"
                              aria-level="1"
                              collapsable={false}
                              numberOfLines={1}
                              style={
                                {
                                  "color": "white",
                                  "fontSize": 17,
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              ホーム
                            </Text>
                          </View>
                          <View
                            collapsable={false}
                            pointerEvents="box-none"
                            style={
                              {
                                "alignItems": "flex-end",
                                "flexBasis": 0,
                                "flexGrow": 1,
                                "justifyContent": "center",
                                "marginEnd": 0,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </RNSScreen>
              </RNSScreenNavigationContainer>
              <View
                collapsable={false}
                onLayout={[Function]}
                pointerEvents="auto"
                style={
                  {
                    "backgroundColor": "white",
                    "borderTopColor": "white",
                    "borderTopWidth": 0.5,
                    "bottom": 0,
                    "elevation": 8,
                    "height": 49,
                    "left": 0,
                    "paddingBottom": 0,
                    "paddingHorizontal": 0,
                    "position": null,
                    "right": 0,
                    "transform": [
                      {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  accessibilityRole="tablist"
                  style={
                    {
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessibilityLabel="ホーム, tab, 1 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": true,
                      }
                    }
                    accessibilityStates={
                      [
                        "selected",
                      ]
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#0075c2"
                          name="home"
                          size={25}
                        />
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#80BAE1"
                          name="home-outline"
                          size={25}
                        />
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "#0075c2",
                          },
                          {
                            "fontSize": 10,
                          },
                          undefined,
                        ]
                      }
                    >
                      ホーム
                    </Text>
                  </View>
                  <View
                    accessibilityLabel="アーティスト, tab, 2 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#0075c2"
                          name="people"
                          size={25}
                        />
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#80BAE1"
                          name="people-outline"
                          size={25}
                        />
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "#80BAE1",
                          },
                          {
                            "fontSize": 10,
                          },
                          undefined,
                        ]
                      }
                    >
                      アーティスト
                    </Text>
                  </View>
                  <View
                    accessibilityLabel="ファンクラブ, tab, 3 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#0075c2"
                          name="calendar"
                          size={25}
                        />
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#80BAE1"
                          name="calendar-outline"
                          size={25}
                        />
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "#80BAE1",
                          },
                          {
                            "fontSize": 10,
                          },
                          undefined,
                        ]
                      }
                    >
                      ファンクラブ
                    </Text>
                  </View>
                  <View
                    accessibilityLabel="グッズ, tab, 4 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#0075c2"
                          name="cart"
                          size={25}
                        />
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#80BAE1"
                          name="cart-outline"
                          size={25}
                        />
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "#80BAE1",
                          },
                          {
                            "fontSize": 10,
                          },
                          undefined,
                        ]
                      }
                    >
                      グッズ
                    </Text>
                  </View>
                  <View
                    accessibilityLabel="設定, tab, 5 of 5"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": false,
                      }
                    }
                    accessibilityStates={[]}
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flex": 1,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "flexDirection": "column",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 0,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#0075c2"
                          name="settings"
                          size={25}
                        />
                      </View>
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "height": "100%",
                              "justifyContent": "center",
                              "minWidth": 25,
                              "position": "absolute",
                              "width": "100%",
                            },
                            {
                              "opacity": 1,
                            },
                          ]
                        }
                      >
                        <Ionicons
                          color="#80BAE1"
                          name="settings-outline"
                          size={25}
                        />
                      </View>
                    </View>
                    <Text
                      numberOfLines={1}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "textAlign": "center",
                          },
                          {
                            "color": "#80BAE1",
                          },
                          {
                            "fontSize": 10,
                          },
                          undefined,
                        ]
                      }
                    >
                      設定
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
